pkgs/dart_mcp_server/lib/src/utils/cli_utils.dart

```diff
@@ -224,7 +225,10 @@ Future<CallToolResult> runCommandInRoot(
   final result = await processManager.run(
     commandWithPaths,
     workingDirectory: workingDir.path,
-    runInShell: true,
+    runInShell:
+        // Required when running .bat files on windows, but otherwise should
+        // be avoided.
+        io.Platform.isWindows && commandWithPaths.first.endsWith('.bat'),
```

Comment:
Are you doing any escaping of the command/args before here, or does that just happen inside `processManager` when `runInShell` is true? (you probably don't want any escaping when this is false).

You might also want to include `.cmd` for completeness (I didn't in Dart-Code because I know the only Windows shell scripts are `flutter.bat`).

I don't know how this works for non-Windows. I presume it you use a shell script that has a shebang line it'll just spawn the shell anyway (Dart-Code never uses shell execute for non-Windows).

Reply:
> Are you doing any escaping of the command/args before here, or does that just happen inside `processManager` when `runInShell` is true? (you probably don't want any escaping when this is false).

Nothing is happening before here